compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -1001,4 +1007,21 @@ private void validateType(SyntaxNodeAnalysisContext ctx, Node node, String type)
                         DiagnosticsCodes.PERSIST_121.getSeverity());
         }
     }
+
+    private boolean validateIfClosedRecord(TypeDefinitionNode typeDefinitionNode) {
+        RecordTypeDescriptorNode recordTypeDescriptorNode =
+                ((RecordTypeDescriptorNode) typeDefinitionNode.typeDescriptor());
+        boolean opened = false;
```

Comment:
Any need to go through all child nodes? RecordTypeDescriptorNode should have specific children node indicating start and end. Identify it and check.
Can there be cases where record start and end can be different?

Reply:
Seems to be that the second child (index 1) is `{|` while the final child is `|}`. Improved the logic as follows in [c032ced](https://github.com/ballerina-platform/module-ballerina-persist/pull/70/commits/c032cedf4deadd70efc7b3ae83ccef7348d629e5)

```
    private boolean isClosedRecord(TypeDefinitionNode typeDefinitionNode) {
        RecordTypeDescriptorNode recordTypeDescriptorNode =
                ((RecordTypeDescriptorNode) typeDefinitionNode.typeDescriptor());
        ChildNodeList children = recordTypeDescriptorNode.children();

        // The second child should be `{|` and the final child should be `|}`
        return children.get(1).kind() == SyntaxKind.OPEN_BRACE_PIPE_TOKEN &&
               children.get(children.size() - 1).kind() == SyntaxKind.CLOSE_BRACE_PIPE_TOKEN;
    }
```

Reply:
Use bodybodyStartDelimeter() https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/RecordTypeDescriptorNode.java#L40

Also since the analysis is done after we check for compile errors, https://github.com/ballerina-platform/module-ballerina-persist/blob/main/compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java#L121 IMO there is no need to check both delimeters

Reply:
Do not use indexes if the API there to already expose this. This way we dont have to worry about changing indexes

Reply:
Fixed in [17dd77a](https://github.com/ballerina-platform/module-ballerina-persist/pull/70/commits/17dd77affcc2d21a507c8b19b68907ea5b2166d4)